src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -117,6 +120,17 @@ internal void Update(long number)
                         break;
                     }
             }
+
+            // There is a race with SnapShot:
```

Comment:
minor: "s/SnapShot/Snapshot/g" (it's one word)

Reply:
I guess we could rename the Snapshot method as well.

Reply:
will do in next PR